### PR TITLE
Clarify docs for `shuffle` arg of `model.fit_generator`

### PR DESCRIPTION
**Pull request by colllin**

### CODE REVIEW

File: keras/engine/training.py [hunk 1]
```diff
@@ -1893,9 +1893,11 @@ def fit_generator(self, generator,
...
                 as they can't be passed
                 easily to children processes.
-            shuffle: Whether to shuffle the data at the beginning of each
-                epoch. Only used with instances of `Sequence` (
-                keras.utils.Sequence).
+            shuffle: Whether to shuffle the order of the batches at 
+                the beginning of each epoch. Only used with instances 
+                of `Sequence` (keras.utils.Sequence). To shuffle the
+                data itself, your `Sequence` should implement 
+                `on_epoch_end`.
```
Comment (fchollet) [inline keras/engine/training.py]: That last sentence will be very difficult to understand for people who don't have further context.

Comment (colllin) [reply keras/engine/training.py]: Thank you for the review & feedback.

The problem is that the `shuffle` option here is misleading. I expected it to behave similarly to the `shuffle` option in `model.fit(...)`, but of course it can't, because it doesn't have access to shuffle the generator's underlying data.  I was trying to improve the situation with documentation.  

Another idea might be to change it `shuffle` from accepting `True` or `False` to accepting `"batch"` or `None`, which I believe is similar in usage & meaning to the `shuffle` option in `model.fit(...)`.  

Or I can make another attempt at updating the documentation, if you think that's a better solution.  Or we can close this and forget about it!

Comment (fchollet) [reply keras/engine/training.py]: "batch shuffling" has a different meaning (shuffling data within mini-batches, but not globally). I think the change in docstring you propose is fine as long as the last sentence is removed.

File: keras/engine/training.py [hunk 2]
```diff
@@ -1893,9 +1893,9 @@ def fit_generator(self, generator,
                 non picklable arguments to the generator
                 as they can't be passed
                 easily to children processes.
-            shuffle: Whether to shuffle the data at the beginning of each
-                epoch. Only used with instances of `Sequence` (
-                keras.utils.Sequence).
+            shuffle: Whether to shuffle the order of the batches at
```
Comment (MatthiasWinkelmann) [inline keras/engine/training.py]: The inserted ```order of``` is less precise than the original. It is the batches that are shuffled, not their order. The order is randomised.

Comment (colllin) [reply keras/engine/training.py]: @MatthiasWinkelmann I think that contradicts what @fchollet said, but I may have confused him.

I think I've sorted (shuffled?) it out.

The only place `fit_generator` uses its `shuffle` argument is where it passes it into the constructor for `OrderedEnqueuer`:
https://github.com/fchollet/keras/blob/master/keras/engine/training.py#L2002

The only place `OrderedEnqueuer` uses its constructor's `shuffle` argument is where it looks like it's shuffling the list of indices which it will use to access the underlying sequence:
https://github.com/fchollet/keras/blob/9c79e01a61b371820d04e9cc52487fc0eb5d0312/keras/utils/data_utils.py#L490

If that's doing what I think it is, then it's shuffling the order of the batches, not the data within each mini-batch, though that could be a good addition for parity with `model.fit(..., shuffle="batch")`.
